libsolidity/interface/CompilerStack.cpp

```diff
@@ -1008,7 +1012,9 @@ Json CompilerStack::yulIROptimizedAst(std::string const& _contractName) const
 
 	// NOTE: Intentionally not using LazyInit. The artifact can get very large and we don't want to
 	// keep it around when compiling a large project containing many contracts.
-	return loadGeneratedIR(contract(_contractName).yulIROptimized).astJson();
+	auto const& currentContract = contract(_contractName);
```

Comment:
This is one of the cases where `auto` hurts readability IMO. Our naming does not usually distinguish it clearly and `contract` may very well refer to `CompilerStack::Contract`, `ContractDefinition` or even a `string`, depending on who wrote the code. It's much nicer to have this clarified with a type whenever possible.

```suggestion
	Contract const& currentContract = contract(_contractName);
```